### PR TITLE
#60: Classes representing table and its indexes

### LINKED ISSUE #60: Classes representing table and its indexes
## Background
Tables need to be represented within the application for further analysis and eventually change SQL entries

## Feature
Design a class that will represent a DB table, its columns and its indexes (no other DB object in scope for now).
Even columns have limited scope.

### PR DESCRIPTION
**Pull request by benedeki**
Closes #60

### CODE REVIEW

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTable.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package za.co.absa.ultet.dbitems
+
+case class DBTable (
+                   tableName: String,
+                   description: Option[String],
+                   primaryDBName: String,
+                   owner: String,
+                   columns: Seq[DBTableColumn] = Seq.empty,
```
Comment (lsulak) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTable.scala:24]: let's write a check that the column can't be there twice or more and that this seq is not empty

Comment (lsulak) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTable.scala:24]: some checks needed for other fields as well

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableIndex.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package za.co.absa.ultet.dbitems
+
+case class DBTableIndex (
+                        indexName: String,
+                        tableName: String,
+                        indexBy: Seq[String],
+                        unique: Boolean = false,
+                        ascending: Boolean = true,
```
Comment (lsulak) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableIndex.scala:24]:
```suggestion
                        ascendingOrder: Boolean = true,
```

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTable.scala (hunk 1 above)
Comment (benedeki) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTable.scala:27]: Yup, in later stages.

File: ultet/src/main/scala/za/co/absa/ultet/sqlentry/SQLEntry.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.sqlentry
+
+trait SQLEntry {
```
Comment (lsulak) [inline ultet/src/main/scala/za/co/absa/ultet/sqlentry/SQLEntry.scala:19]: this can be even deleted - there is a new one and it already has the enum in it

Comment (lsulak) [reply ultet/src/main/scala/za/co/absa/ultet/sqlentry/SQLEntry.scala:19]: will be available after merge conflict resolution

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableMember.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                           tableName: String,
+                           indexBy: Seq[String],
+                           unique: Boolean = false,
+                           ascending: Boolean = true,
+                           nullsFirstOverride: Option[Boolean] = None,
+                           nullsDistinct: Boolean = true
+                         )
+    extends DBTableMember {
+
+    if (indexBy.map(_.toLowerCase).distinct.length
```
Comment (lsulak) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableMember.scala:33]: unfinished?

**Comment by lsulak** [comment]
Found few things but LGTM otherwise

**Review by dk1844 — APPROVED** [review]

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableMember.scala (hunk 4 above)
Comment (lsulak) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBTableMember.scala:33]:
```suggestion
```

**Review by lsulak — APPROVED** [review]

**Pull request merged**